### PR TITLE
infra create and update v2

### PR DESCRIPTION
**Pull request by rocktavious**
This change proposes a different signature for the create and update infra methods because the ones based on the API are not intuitive for customers when I put them into use and IMO this provides a much better UX when using the library.

### CODE REVIEW

File: cache.go [hunk 1]
```diff
@@ -101,7 +101,7 @@ func (c *Cacher) TryGetInfrastructureSchema(alias string) (*InfrastructureResour
 }
 
 func (c *Cacher) doCacheTiers(client *Client) {
-	log.Info().Msg("Caching 'Tier' lookup table from API ...")
+	log.Debug().Msg("Caching 'Tier' lookup table from API ...")
```
Comment (rocktavious) [inline cache.go:104]: These info logs were screwing things up in the CLI

File: infra.go [hunk 2]
```diff
@@ -56,7 +56,36 @@ type InfrastructureResourceInput struct {
 	Data         JSON                                 `json:"data,omitempty" yaml:"data" scalar:"true"`
 }
 
-func (client *Client) CreateInfrastructure(input InfrastructureResourceInput) (*InfrastructureResource, error) {
+type InfraProviderInput struct {
```
Comment (rocktavious) [inline infra.go:59]: These new input types don't exist in our API but they provide a WAY better customer UX when using the library.  See the code in the CLI and Terraform as implementation examples.

File: infra.go [hunk 3]
```diff
@@ -56,7 +56,36 @@ type InfrastructureResourceInput struct {
...
+
+type InfraInput struct {
+	Schema   string              `json:"schema" yaml:"schema"`
+	Owner    *ID                 `json:"owner" yaml:"owner"`
+	Provider *InfraProviderInput `json:"provider" yaml:"provider"`
+	Data     JSON                `json:"data" yaml:"data" scalar:"true"`
+}
+
+func (client *Client) CreateInfrastructure(input InfraInput) (*InfrastructureResource, error) {
+	i := InfrastructureResourceInput{
```
Comment (rocktavious) [inline infra.go:74]: We do this translation layer in each function now.

**Review by davidbloss — APPROVED** [review]
I'm liking the cli UX improvements 🤩 
LGTM! 🎉

**Pull request merged**